feat(treasury): index governor votes and tallies

rickstaa opened the pull request:
This pull erquest ensures LivepeerGovernor votes are tracked with per-voter state and proposal-level aggregates.

jmulq commented on schema.graphql:
```diff
@@ -371,6 +374,40 @@ type TreasuryProposal @entity {
+  totalVotes: BigDecimal!
+}
+
+enum TreasuryVoteSupport @entity {
```
Has this been deployed and is it syncing? If it is I'm pretty confused. 

I didn't think `enum` types could be entities. I would remove the `@entity` annotation.

rickstaa replied:
Good catch! You're right (see the enum documentation: [https://thegraph.com/docs/en/subgraphs/guides/enums/#what-are-enums](https://thegraph.com/docs/en/subgraphs/guides/enums/#what-are-enums)).

The subgraph was deployed, but it appears it previously failed earlier in the process due to a missing ABI. I've just fixed that issue [here](https://github.com/livepeer/subgraph/pull/175/changes/5f5dc620fa9f19686580584e7af27603deb2d087)

rickstaa replied:
@jmulq Looks like we do this in more places created an issue for it: https://github.com/livepeer/subgraph/issues/177.

jmulq reviewed (changes requested):
Left a few comments. Nothing major but I'd definitely look at changing stuff around the `enum` comments. Has this been deployed and is it syncing without errors?

rickstaa commented on src/mappings/treasury.ts:
```diff
@@ -1,16 +1,147 @@
+    voter,
+    event.block.timestamp.toI32()
+  );
+  const round = createOrLoadRound(getBlockNum());
```
All occurrences of `getBlockNum` originally used `event.block.number` but was changed in https://github.com/livepeer/subgraph/pull/19 to I think contain it to our contracts. However under the hood they both are equivalent (see https://github.com/livepeer/protocol/blob/d03671e2e159bbc856feace83371876cf25124e3/contracts/rounds/RoundsManager.sol#L123). Maybe we create an issue and tackle this after your audit to see what makes more sense using `event.block.number` directly or renaming the function? I created https://github.com/livepeer/subgraph/issues/176 to track this.

rickstaa commented on src/mappings/treasury.ts:
```diff
@@ -1,16 +1,147 @@
+    vote.voter = account.id;
+  }
+
+  vote.support = supportLabel;
```
@jmulq I looked into this while working on the PR. Per the Graph docs on enums https://thegraph.com/docs/en/subgraphs/guides/enums/, enum fields in mappings are set using their string values, and the generated AssemblyScript entities expose them as `string`, not typed enums. AssemblyScript/codegen also don't currently support string enums (see https://github.com/AssemblyScript/assemblyscript/issues/560).

I might be missing something, but if not, would https://github.com/livepeer/subgraph/pull/175/changes/d39ea1a8c8f8205a2fd53e698887f802077b0368 be an acceptable workaround for now?

rickstaa commented:
@jmulq thanks again for the thorough review. I’ve addressed most of the comments. The only one I couldn’t fully resolve yet is the enum-related point. Could you take a look at the proposed solution and let me know if it works for you, or if you see a better approach? Thanks in advance.

jmulq reviewed (approved):
Approving this so long as the Livepeer team are happy with the decided implementation for the enum logic. Feel free to revert it back to the prior code, or use the implemented workaround.
